msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs

```diff
@@ -20,17 +20,37 @@ public class FilterStaticFrameworks : XamarinTask, ITaskCallback {
 		[Output]
 		public ITaskItem []? FrameworkToPublish { get; set; }
 
-		static string GetFrameworkExecutablePath (string frameworkPath, TaskLoggingHelper? log = null)
+		static string [] GetFrameworkInfoPlistPaths (string frameworkPath, ApplePlatform platform)
+		{
+			switch (platform) {
+			case ApplePlatform.iOS:
+			case ApplePlatform.TVOS:
+				return new string [] {
+					Path.Combine (frameworkPath, "Info.plist")
+				};
+			case ApplePlatform.MacOSX:
+			case ApplePlatform.MacCatalyst:
+				return new string [] {
+					Path.Combine (frameworkPath, "Resources", "Info.plist"), // Check symlink first
+					Path.Combine (frameworkPath, "Versions", "A", "Resources", "Info.plist") // Fallback to actual location
+				};
+			default:
+				// For unknown platforms, try both common locations
+				return new string [] {
+					Path.Combine (frameworkPath, "Info.plist"),
+					Path.Combine (frameworkPath, "Resources", "Info.plist")
+				};
```

Comment:
Throw an exception instead (look for examples in sibling files for InvalidOperationException with MSBStrings.InvalidPlatform as the message).

This also means this function will only ever return a single value, so it can be simplified to do just that.